File: programs/assignment1/ProgrammingAssignment1/src/programmingassignment1/Janitor.java
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 1
// This is my own work; I will not post

package programmingassignment1;

/**
 * 
 * @author Adrian Davila
 */
public class Janitor extends Employee{
    String department;
    private char sweeping;

    Janitor(){
        department = "";
        sweeping = 'N';
    }
    Janitor(char newRole , String newName , int newID , String dept , char cleaning){
        super(newRole , newName , newID);
        department = dept;
        sweeping = cleaning;
        addEmployee((Employee)this);
    }

    public void deleteEmployee(){
        deleteEmployee(getName() , getRole());
    }
    
    // Mutators
    public void setDept(String newDept){
        department = newDept;
    }
    public void setworkStat(char workStat){
        sweeping = workStat;
    }
    // Accessors
    public String getDept(){
        return department;
    }
    public char getworkStat(){
        return sweeping;
    }

    @Override
    public String toString(){
        return (super.toString() + " "  + department + " " + sweeping);
    }
    @Override
    public String toString(boolean t){
        return (super.toString() + " Department: "  + department + " Sweeping: " + sweeping);
    }
    @Override
    public boolean equals(Employee obj){
        if(obj == null){
            return false;
        }else if(getClass() != obj.getClass()){
            return false;
        }else{
            Janitor otherJan = (Janitor)obj;
            return (getName().equals(otherJan.getName()) && getRole() == otherJan.getRole());
        }
    }
}

File: programs/assignment3/ProgrammingAssignment3/ProgrammingAssignment3/Janitor.cpp
// COSC 1320 Summer 2015
// Name: Adrian
// Programming Assignment 3
// This is my own work; I will not post

#pragma once
#include <string>
#include "Administrator.h"
#include "Janitor.h"
using namespace std;

Janitor::Janitor() : Administrator('j' , "none" , 0 , "Maintenance"){
	sweeping = 'N';
}


Janitor::~Janitor(void)
{
}
Janitor::Janitor(char newRole, string newName, int newID, string newDept, char busyState):Administrator(newRole , newName , newID , newDept){
	sweeping = busyState;
}
char Janitor::getSweeping(){
	return sweeping;
}
void Janitor::setSweeping(char busyState){
	sweeping = busyState;
}
string Janitor::toString(){
	string str = " ";
	str += sweeping;
	return (Administrator::toString() + str);
}
bool Janitor::equals(HospitalEmployee obj){
	if(obj.getID() == getID() && obj.getName() == getName() && obj.getRole() == getRole()){
		return true;
	} else {
		return false;
	}
}